src/app/login-menu/login-menu/login-menu.component.ts

```diff
@@ -0,0 +1,46 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { RouterModule, Router } from '@angular/router';
+import { ValidationService } from 'src/app/validation.service';
+
+
+@Component({
+  selector: 'app-login-menu',
+  templateUrl: './login-menu.component.html',
+  styleUrls: ['./login-menu.component.scss']
+})
+
+export class LoginMenuComponent implements OnInit {
+
+  mail: string = "";
+  password: string = "";
+  message: string = "";
```

Comment:
Is it necessary to initialize these variables as empty strings instead of just declaring them like message : string; ? You seems to overwrite them later anyway.

Also where exactly do password and mail get overwritten? I see this component using them but not overriding them.

Oh I almost forgot
![image](https://user-images.githubusercontent.com/43812715/84050751-732db380-a9ae-11ea-841c-e2fd66bcc703.png)